src/js/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,202 @@
+import React, { forwardRef, useContext, useState } from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { Box } from '../Box';
+import { Nav } from '../Nav';
+import { PageControl } from './PageControl';
+
+const StyledPaginationContainer = styled(Box)`
+  ${props =>
+    props.theme.pagination.container && props.theme.pagination.container.extend}
+`;
+
+const getPageIndices = (begin, end) => {
+  const indices = [];
+  for (let i = begin; i <= end; i += 1) {
+    indices.push(i);
+  }
+  return indices;
+};
+
+const Pagination = forwardRef(
+  (
+    {
+      a11yTitle,
+      numItems,
+      numEdgePages = 1, // number of pages at each edge of page indices
+      // number of page controls in the middle
+      numMiddlePages: numMiddlePagesProp = 3,
+      onChange,
+      page: pageProp,
+      step = 10,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+
+    /* Calculate total number pages */
+    const totalPages = Math.ceil(numItems / step);
+    const [activePage, setActivePage] = useState(
+      Math.min(pageProp, totalPages) || 1,
+    );
+
+    /* Define page indices to display */
+    const beginPages = getPageIndices(1, Math.min(numEdgePages, totalPages));
+    const endPages = getPageIndices(
+      Math.max(totalPages - numEdgePages + 1, numEdgePages + 1),
+      totalPages,
+    );
+
+    let numMiddlePages;
+    if (numMiddlePagesProp < 1) {
+      numMiddlePages = 1;
+      console.warn(
+        // eslint-disable-next-line max-len
+        `Property "numMiddlePages" should not be < 1. One middle page button will be shown. Set "numMiddlePages" >= 1 to remove this warning.`,
+      );
```

Comment:
Is there a way to control the input within propTypes and typescript instead of us placing the warning?

Reply:
I will have a look and see.

Reply:
So for this, we are setting  numberMiddlePages to 1 if they set it to anything less than that. The warning is intended to just alert the user that we are changing something for them. If this warning feels unnecessary, we can just remove it.